plover/gui_qt/dictionaries_widget.py

```diff
@@ -256,3 +263,56 @@ def on_move_dictionaries_down(self):
             return
         self._update_dictionaries(dictionaries)
         self._set_selection(selection)
+
+    def _row_selector(self, position):
+        position = self.table.mapToGlobal(position)
+        menu = QMenu()
+
+        row_action = QAction("Save As...", self)
+        row_action_open = QAction("Open...", self)
+
+        row_action.triggered.connect(self.save_as)
+        row_action_open.triggered.connect(self.on_add_dictionaries)
+
+        menu.addAction(row_action)
+        menu.addAction(row_action_open)
+
+        menu.exec(position)
+
+    def save_as(self):
+        filename = [self._dictionaries[item.row()]
+                        for item in self.table.selectedItems()][0]
+        file_ext = os.path.splitext(filename)[-1].lower()
+        if file_ext in dictionary_formats:
+            selectedFilter = '%s Dictionary (*%s)' % (file_ext.upper().strip('.'), file_ext)
+
+        filters = ';;'.join('%s Dictionary (*%s)' % (ext.upper().strip('.'), ext) for ext in dictionary_formats)
+        new_filename = QFileDialog.getSaveFileName(
+            self, _('Save Dictionary'), filename,
+            filters, selectedFilter
+        )[0]
+
+        if new_filename:
+            basename = os.path.basename(filename)
+            ext_new = os.path.splitext(new_filename)[-1].lower()
+
+            if file_ext == ext_new:
+                shutil.copyfile(filename, new_filename)
+            else:
+                open(new_filename, "w")
+                self.convert_dictionary(filename, new_filename)
+
+            msg = basename + " was saved to " + new_filename + ". Would you like to add the new dictionary to Plover?"
```

Comment:
+            msg = basename + " was saved to " + new_filename + ". Would you like to add the new dictionary to Plover?" 

`msg = '%s was saved to %s\n\nWould you like to add the new dictionary to Plover?' % (file_name, file_directory)`

Mm, also code style dictates that we use `''` for strings, not `""`